File: examples/python-repo/ecommerce/domain/customer/address.py
# DELIBERATE: wildcard import (Ruff F403)
from string import *


class Address:
    def __init__(self, street, city, zip_code, country):
        self._street = street
        self._city = city
        self._zip_code = zip_code
        self._country = country

    @property
    def street(self):
        return self._street

    @property
    def city(self):
        return self._city

    @property
    def zip_code(self):
        return self._zip_code

    @property
    def country(self):
        return self._country

    def formatted(self):
        return f"{self._street}, {self._city} {self._zip_code}, {self._country}"

File: examples/java-repo/src/main/java/com/example/ecommerce/domain/customer/Address.java
package com.example.ecommerce.domain.customer;

public class Address {

    private final String street;
    private final String city;
    private final String zipCode;
    private final String country;

    public Address(String street, String city, String zipCode, String country) {
        // DELIBERATE: empty catch block (PMD EmptyCatchBlock)
        try {
            if (zipCode != null) {
                Integer.parseInt(zipCode.replaceAll("-", ""));
            }
        } catch (NumberFormatException e) {
        }

        this.street = street;
        this.city = city;
        this.zipCode = zipCode;
        this.country = country;
    }

    public String getStreet() {
        return street;
    }

    public String getCity() {
        return city;
    }

    public String getZipCode() {
        return zipCode;
    }

    public String getCountry() {
        return country;
    }

    public String formatted() {
        return street + ", " + city + " " + zipCode + ", " + country;
    }
}
